backend/app/interfaces/email_service.py

```diff
@@ -1,4 +1,38 @@
+import json
 from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from enum import Enum
+from typing import Generic, TypeVar
+
+T = TypeVar("T")
+
+
+class TemplateData(ABC):
+    def get_formatted_string(self) -> str:
+        class_dict = self.__dict__
+        try:
+            formatted_string = json.dumps(class_dict)  # Try to convert to a JSON string
+        except (TypeError, ValueError) as e:
+            # Handle errors and return a message instead
+            return f"Error in converting data to JSON: {e}"
```

Comment:
If we're planning on using this as the basis of other template data, I think we should propagate the error instead of gracefully failing, especially if this is being sent by email.